### PR TITLE
Use Vitepress instead of VuePress

### LINKED ISSUE #782: Switch to vitepress
VitePress 1.0 was released.

Current version of this library uses VuePress for generating document.
VuePress is no longer maintained(maybe). And VitePress is the successor of the VuePress.

I want to use VitePress instead of VuePress!

### PR DESCRIPTION
**Pull request by tokuhirom**
VuePress is no longer maintained. And the authors says:

> VuePress is now in maintenance mode. For a next-gen Vue-based SSG built on top of Vue 3 + Vite, check out [VitePress](https://vitepress.vuejs.org/).
> https://github.com/vuejs/vuepress

Close #782

### CODE REVIEW

File: docs/.gitignore [hunk 1]
```diff
@@ -1,3 +1,7 @@
 /.vuepress/.cache/
 /.vuepress/.temp/
 /apidocs/
+
+/CONTRIBUTING.md
+/index.md
```
Comment (tokuhirom) [inline docs/.gitignore:6]: Formely, it's just a symlink. But vitepress doesn't process symlink correctly. We need to copy the file instead.

File: docs/.vitepress/config.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
+const utils = require('./utils.ts');
+
+utils.copyFile('README.md', 'index.md');
+utils.copyFile('CONTRIBUTING.md', 'CONTRIBUTING.md');
```
Comment (tokuhirom) [inline docs/.vitepress/config.ts:4]: Previous implementation do this by symlink. But vitepress doesn't support it.
As a result, we need to copy the file instead.

File: docs/.vitepress/config.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+utils.rewriteFile('../apidocs/README.md', /\(CONTRIBUTING.md\)/g, '(../CONTRIBUTING.md)');
```
Comment (tokuhirom) [inline docs/.vitepress/config.ts:6]: typedoc copies the `/README.md` to `docs/apidocs/README.md`.  And it contains the link to `CONTRIBUTING.md`. But there's no document at `docs/apidocs/CONTRIBUTING.md`. We need to fix the link.

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -42,6 +42,8 @@ jobs:
         python3 generate-code.py
     - name: Test Project
       run: export NODE_OPTIONS=--max-old-space-size=6144; npm test
+    - name: Test building apidocs
```
Comment (Yang-33) [inline .github/workflows/test.yml:45]: 😲

**Review by Yang-33 — APPROVED** [review]
Thank you!

**Pull request merged**